<launch>
  <arg name="use_provided_manager" default="false" />
  <arg name="provided_manager" default="camera/camera_nodelet_manager"/>
  <arg name="nodelet_manager" default="egocylindrical_nodelet_manager" unless="$(arg use_provided_manager)" />
  <arg name="nodelet_manager" default="$(arg provided_manager)" if="$(arg use_provided_manager)" />

  <arg name="depth_image" default="/camera/depth/image_raw"/>


  <node name="$(arg nodelet_manager)" pkg="nodelet" type="nodelet" args="manager" output="screen"/>
  
  <node name="depth_to_egocylindrical_nodelet" pkg="nodelet" type="nodelet" args="load egocylindrical/egocylindrical_propagator_nodelet $(arg nodelet_manager)" >
    <remap from="/camera/depth/image_raw" to="$(arg depth_image)"/>
  </node>
  
  <node name="egocylindrical_to_pointcloud_nodelet" pkg="nodelet" type="nodelet" args="load egocylindrical/point_cloud_nodelet $(arg nodelet_manager)" />

  <node name="egocylindrical_to_range_image_nodelet" pkg="nodelet" type="nodelet" args="load egocylindrical/range_image_nodelet $(arg nodelet_manager)"/>
 
</launch>
